tests/profiling/collector/test_memalloc.py

```diff
@@ -64,171 +69,249 @@ def _pre_allocate_1k():
 
 def test_iter_events():
     max_nframe = 32
-    _memalloc.start(max_nframe, 10000, 512 * 1024)
-    _allocate_1k()
-    events, count, alloc_count = _memalloc.iter_events()
-    _memalloc.stop()
-
-    assert count >= 1000
-    # Watchout: if we dropped samples the test will likely fail
-
-    object_count = 0
-    for (stack, nframe, thread_id), size, domain in events:
-        assert domain == "object"
-        assert 0 < len(stack) <= max_nframe
-        assert nframe >= len(stack)
-        last_call = stack[0]
-        assert size >= 1  # size depends on the object size
-        if last_call == DDFrame(
-            __file__,
-            _ALLOC_LINE_NUMBER,
-            "<listcomp>" if sys.version_info < (3, 12) else "_allocate_1k",
-            "",
-        ):
-            assert thread_id == threading.main_thread().ident
-            if sys.version_info < (3, 12):
-                assert stack[1] == (__file__, _ALLOC_LINE_NUMBER, "_allocate_1k", "")
-            object_count += 1
-
-    assert object_count >= 1000
+    collector = memalloc.MemoryCollector(max_nframe=max_nframe, _max_events=10000, heap_sample_size=64)
+    collector._start_service()
+    try:
+        _allocate_1k()
+        samples = collector.test_snapshot()
+        alloc_samples = [s for s in samples if s.alloc_size > 0]
+
+        total_alloc_count = sum(s.count for s in alloc_samples)
+
+        assert total_alloc_count >= 1000
+        # Watchout: if we dropped samples the test will likely fail
+
+        object_count = 0
+        for sample in alloc_samples:
+            stack = sample.frames
+            thread_id = sample.thread_id
+            size = sample.alloc_size
+
+            assert 0 < len(stack) <= max_nframe
+            assert size >= 1  # size depends on the object size
+            last_call = stack[0]
+            if last_call == DDFrame(
+                __file__,
+                _ALLOC_LINE_NUMBER,
+                "<listcomp>" if sys.version_info < (3, 12) else "_allocate_1k",
+                "",
+            ):
+                assert thread_id == threading.main_thread().ident
+                if sys.version_info < (3, 12) and len(stack) > 1:
+                    assert stack[1] == DDFrame(__file__, _ALLOC_LINE_NUMBER, "_allocate_1k", "")
+                object_count += sample.count
+
+        assert object_count >= 1000
+    finally:
+        collector._stop_service()
 
 
 def test_iter_events_dropped():
     max_nframe = 32
-    _memalloc.start(max_nframe, 100, 512 * 1024)
-    _allocate_1k()
-    events, count, alloc_count = _memalloc.iter_events()
-    _memalloc.stop()
+    collector = memalloc.MemoryCollector(max_nframe=max_nframe, _max_events=100, heap_sample_size=64)
+    collector._start_service()
+    try:
```

Comment:
We can do `with collector` instead of a `try` block and delete the `finally` block